dotCMS/src/main/java/com/liferay/portal/ejb/UserPersistence.java

```diff
@@ -231,31 +240,28 @@
 		}
 	}
 
+	@CloseDBIfOpened
 	protected com.liferay.portal.model.User findByPrimaryKey(String userId)
-		throws NoSuchUserException, SystemException {
+			throws NoSuchUserException, SystemException {
 		com.liferay.portal.model.User user = UserPool.get(userId);
-		Session session = null;
 
 		try {
 			if (user == null) {
-				session = openSession();
-
-				UserHBM userHBM = (UserHBM)session.load(UserHBM.class, userId);
-				user = UserHBMUtil.model(userHBM);
+				DotConnect dc = new DotConnect();
+				dc.setSQL(SEARCH_USER_BY_ID);
+				dc.addParam(userId);
+				List<Map<String, Object>> list = dc.loadObjectResults();
+				if(list.isEmpty()) {
+					throw new NoSuchUserException(userId.toString());
+				}else{
+					user = userTransformer.fromMap(list.get(0));
+				}
 			}
 
 			return user;
 		}
-		catch (HibernateException he) {
-			if (he instanceof ObjectNotFoundException) {
-				throw new NoSuchUserException(userId.toString());
-			}
-			else {
-				throw new SystemException(he);
-			}
-		}
-		finally {
-			HibernateUtil.closeSession(session);
+		catch(DotDataException e){

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1166)